File: algorithms/java/validParentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> myStack = new Stack<>();

        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '(' || s.charAt(i) == '{' || s.charAt(i) == '['){
                myStack.add(s.charAt(i));
            }
            else{
                if(myStack.size() == 0){
                    return false;
                }
                switch(s.charAt(i)){
                    case ')': 
                        if(myStack.peek() == '('){
                            myStack.pop();
                        }
                        else
                            return false;        
                        break;
                    case '}':
                        if(myStack.peek() == '{'){
                            myStack.pop();
                        }
                        else
                            return false;
                        break;
                    case ']':
                        if(myStack.peek() == '['){
                            myStack.pop();
                        }
                        else
                            return false;
                        break;
                    default:
                        break;
                }
            }      
        }
        if(myStack.size() == 0){
            return true;
        }
        return false;
    }
}

File: algorithms/python/validParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        myStack = list()

        for char in s:
            if char == '(' or char == '[' or char == '{':
                myStack.append(char)
            else:
                if len(myStack) == 0:
                    return False
                if char == ')' and myStack[len(myStack) - 1] == '(':
                    myStack.pop()
                elif char == ']' and myStack[len(myStack) - 1] == '[':
                    myStack.pop()
                elif char == '}' and myStack[len(myStack) - 1] == '{':
                    myStack.pop()
                else:
                    return False
        if len(myStack) == 0:
            return True
        return False
                